crates/oxc_linter/src/rules/eslint/no_obj_calls.rs

```diff
@@ -61,44 +65,111 @@ declare_oxc_lint! {
     correctness,
 }
 
-fn is_global_obj<'a>(str: impl PartialEq<&'a str>) -> bool {
-    NON_CALLABLE_GLOBALS.iter().any(|&n| str == n)
+fn is_global_obj<'a>(str: &impl PartialEq<&'a str>) -> bool {
+    NON_CALLABLE_GLOBALS.iter().any(|&n| str == &n)
+}
+
+fn global_this_member(expr: &oxc_allocator::Box<'_, MemberExpression<'_>>) -> Option<Atom> {
+    if let Expression::Identifier(static_ident) = expr.object() &&
+    static_ident.name == GLOBAL_THIS &&
+    let Some(static_member) = expr.static_property_name() {
+        return Some(static_member.into())
+    } else {
+        return None
+    }
+}
+
+fn resolve_global_binding<'a, 'b: 'a>(
+    ident: &oxc_allocator::Box<'a, IdentifierReference>,
+    scope_id: ScopeId,
+    ctx: &LintContext<'a>,
+) -> Option<Atom> {
+    if ctx.semantic().is_reference_to_global_variable(ident) {
+        return Some(ident.name.clone());
+    } else {
+        let scope = ctx.scopes();
```

Comment:
It feels like we need this due not having `ReferenceId` on the identifier reference right? Otherwise we can get its referenced symbol directly from the symbol table.

Reply:
https://github.com/Boshen/oxc/issues/510

Reply:
This is correct. I believe line 96 is related.